### PR TITLE
[WIP] Refactoring ukub log node with telemetry

### PR DESCRIPTION
**Pull request by cnalley**
:construction: 
Done:
Refactored original ukub log node code.
Added basic telemetry storage functions for creating filenames and log entries.
Refactored and reworked SD card IO functions.
Added inter subsystem CSP receive task.
Added telemetry system usage between task.

TODO:
Code cleanup and review.
CSP task (in the telemetry system) must be running at priority 1.
Another PR against the telemetry system will need to allow configuration of the telemetry task priority.
telemetry_get_subs and telemetry_rx_task in telemetry.h need to be priority 1 and csp_route_start_task in communications.h needs to be priority 2.

:construction: 

/cc @kubostech/devs

### CODE REVIEW

**Comment by kyleparrott** [comment]
So now that I've hijacked this PR from @cnalley it should be ready for some review. 

This should be the full finished code for the log node. It's currently subscribing to all of the telemetry publishers and should be storing the data to the sd card in a single file per publisher.

File: source/telemetry_storage.c [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        }
+    return len; }
+
+    if(packet.source.data_type == TELEMETRY_TYPE_FLOAT) {
+        len = snprintf(data_buf_ptr, DATA_BUFFER_SIZE, "%u,%f\r\n", packet.timestamp, packet.data.f);
+        if(len < 0 || len >= DATA_BUFFER_SIZE) {
+            printf("Data char limit exceeded for float packet. Have %d, need %d + \\0\n", DATA_BUFFER_SIZE, len);
+            return 0;
+        }
+    return len;
```
Comment (plauche) [inline source/telemetry_storage.c]: nit: Formatting/indentation is a bit inconsistent in this function

File: source/telemetry_storage.c [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    }
+}
```
Comment (plauche) [inline source/telemetry_storage.c]: This function could just return len once at the bottom instead of inside each if block.

File: source/main.c [hunk 3]
```diff
@@ -18,148 +18,151 @@
...
     k_gpio_init(K_LED_BLUE, K_GPIO_OUTPUT, K_GPIO_PULL_NONE);
-    k_gpio_init(K_BUTTON_0, K_GPIO_INPUT, K_GPIO_PULL_NONE);
 
-    xTaskCreate(task_logging, "logging", configMINIMAL_STACK_SIZE * 5, NULL, 2, NULL);
+    /* Configure and initialize UART bus and baudrate */
+    struct usart_conf conf;
+    char dev = (char)CSP_UART_BUS;
+    conf.device = &dev;
+    conf.baudrate = CSP_UART_BAUDRATE;
+    usart_init(&conf);
```
Comment (plauche) [inline source/main.c]: Should `usart_init`, `csp_kiss_init` and `usart_set_callback` go after `telemetry_init` (which contains `csp_init`)?

File: source/disk.c [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+}
+
+
+/** 
+ * @brief Closes a file and releases the handle.
+ * @param Fil a pointer to the file object structure
+ * @return ret a table of values which (0 being 'okay') is found at
+ *         http://elm-chan.org/fsw/ff/en/rc.html  
+ */
+static uint16_t close_file(FIL *Fil)
```
Comment (plauche) [inline source/disk.c]: This wrapper function might be overkill, since it is just calling `f_close` directly and not really doing anything else.

**Comment by plauche** [comment]
Looks good to merge here :shipit:

**Pull request merged**